### DISCUSSION TITLE
Is there a way to disable payload hashing when uploading an object to s3?

### DISCUSSION THREAD

**Discussion by tweeter0830** [awslabs/aws-sdk-rust#854] [Q&A] [answered]
I noticed really high CPU usage when calling `aws_sdk_s3::client.put_object`. I profiled my code and saw a ton of time being spent in `aws_sign_auth::signer::SigV4Signer::Sign`, specifically `aws_sigv4::http_request::canocial_request::CanonicalRequest::payload_hash`. I know that AWS hashes some/all of the request, but I'm hoping there is some way to avoid hashing the body of the request. I am already sending an md5 hash of the body along.

Thanks!

**Comment by Velfi** [comment, marked as answer]
If you upload with a streaming body, then it'll avoid calculating those hashes. Use the [`read_from` API](https://docs.rs/aws-sdk-s3/latest/aws_sdk_s3/primitives/struct.ByteStream.html#method.read_from) when uploading files.

Let me know if that works for you.

**Reply by jiashiwen** [reply]
![flamegraph_00](https://github.com/awslabs/aws-sdk-rust/assets/5244267/2d140be4-6deb-4c04-9a2f-c65d8e4d447d)
Hi,and I meet a same question
<[Using mutlity part upload cpu is too high](https://github.com/awslabs/aws-sdk-rust/discussions/1085)>
I use ByteStream,but steel a lost of calculate pyload_hash
